src/Functions/FunctionsDecimalArithmetics.cpp

```diff
@@ -0,0 +1,17 @@
+#include <Functions/FunctionsDecimalArithmetics.h>
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+REGISTER_FUNCTION(DivideDecimals)
+{
+    factory.registerFunction<FunctionsDecimalArithmetics<DivideDecimalsImpl>>(Documentation(
+        "Decimal division with given precision. Slower than simple `divide`, but has controlled precision and no sound overflows"));
```

Comment:
The documentation is not enough.
Every function should be in its own .cpp file.

Reply:
Ok, will do it like this next time.